python/tests/unit/arcticdb/version_store/test_resample.py

```diff
@@ -43,18 +47,32 @@ def generic_resample_test_with_empty_buckets(lib, sym, rule, aggregations, date_
 
     assert_frame_equal(expected, received, check_dtype=False)
 
-def generic_resample_test(lib, sym, rule, aggregations, date_range=None, closed=None, label=None, offset=None):
+def generic_resample_test(lib, sym, rule, aggregations, date_range=None, closed=None, label=None, offset=None, origin=None, drop_empty_buckets_for=None):
     # Pandas doesn't have a good date_range equivalent in resample, so just use read for that
     expected = lib.read(sym, date_range=date_range).data
     # Pandas 1.X needs None as the first argument to agg with named aggregators
+
+    pandas_aggregations = {**aggregations, "_bucket_size_": (drop_empty_buckets_for, "count")} if drop_empty_buckets_for else aggregations
+    resample_args = {}
+    if origin:
+        resample_args['origin'] = origin
+    if offset:
+        resample_args['offset'] = offset
+
     if PANDAS_VERSION >= Version("1.1.0"):
-        expected = expected.resample(rule, closed=closed, label=label, offset=offset).agg(None, **aggregations)
+        expected = expected.resample(rule, closed=closed, label=label, **resample_args).agg(None, **pandas_aggregations)
     else:
-        expected = expected.resample(rule, closed=closed, label=label).agg(None, **aggregations)
+        expected = expected.resample(rule, closed=closed, label=label).agg(None, **pandas_aggregations)
+    if drop_empty_buckets_for:
+        expected = expected[expected["_bucket_size_"] > 0]
+        expected.drop(columns=["_bucket_size_"], inplace=True)
```

Comment:
NaN and None count as existing values so I think it's OK. The reason I had to invent this is because the other method removing empty buckets (based on groupby) does take origin.

Reply:
Update: yeah it breaks for float columns with NaNs

Reply:
At least needs a comment to not use it on float columns then?
I'm pretty sure string columns with only Nones/NaNs in a bucket will also produce a count of zero:
https://github.com/man-group/ArcticDB/blob/master/cpp/arcticdb/processing/sorted_aggregation.hpp#L375
https://github.com/man-group/ArcticDB/blob/master/cpp/arcticdb/processing/sorted_aggregation.hpp#L318-L320